python/ray/tune/execution/experiment_state.py

```diff
@@ -182,107 +133,52 @@ def checkpoint(
+            driver_staging_path, self._storage.experiment_fs_path
+        )
+        if launched_sync:
+            self._last_sync_time = time.monotonic()
 
-        if sync_time_taken > self._slow_sync_threshold:
-            logger.warning(
-                "Syncing the experiment checkpoint to cloud took a "
-                f"{sync_time_taken:.2f} seconds. This can be due to a large number "
-                f"of trials, large logfiles, or throttling from the "
-                f"remote storage provider for too frequent syncs.\n"
-                f"If you set `CheckpointConfig.num_to_keep` to a low number, this can "
-                f"trigger frequent syncing. Try increasing the `num_to_keep`. "
-            )
+        if force:
+            wait_for_sync()
 
-        if not synced:
-            return False
+        checkpoint_time_taken = time.monotonic() - checkpoint_time_start
 
-        self._should_force_cloud_sync = False
-        self._trial_num_checkpoints_since_last_sync.clear()
+        # Adjust dynamic checkpointing
+        self._update_auto_checkpoint_time(time_taken=checkpoint_time_taken)
 
-        if now - self._last_sync_time < self._excessive_sync_threshold:
+        time_since_last_sync = time.monotonic() - self._last_sync_time
+        if time_since_last_sync > self._slow_sync_threshold:
             logger.warning(
-                "Experiment checkpoint syncing has been triggered multiple "
-                f"times in the last {self._excessive_sync_threshold} seconds. "
-                "A sync will be triggered whenever a trial has checkpointed "
-                "more than `num_to_keep` times since last sync or if "
-                f"{syncer.sync_period} seconds have passed since last "
-                "sync. If you have set `num_to_keep` in your `CheckpointConfig`, "
-                "consider increasing the checkpoint frequency or keeping more "
-                "checkpoints. You can supress this warning by changing the "
-                "`TUNE_WARN_EXCESSIVE_EXPERIMENT_CHECKPOINT_SYNC_THRESHOLD_S` "
-                "environment variable."
+                "Saving the experiment state (which holds a global view "
+                "of trial statuses and is used to restore the experiment) has already "
+                f"taken {time_since_last_sync:.2f} seconds, which may cause consistency"
+                " issues upon restore if your driver script ungracefully exits.\n"
+                "This could be due to a large number of trials, "
+                "large logfiles from lots of reported metrics, or throttling from the "
+                "remote storage if uploading too frequently.\n"
+                "You can increase the number of seconds between experiment syncs "
+                "by setting the environment variable TUNE_GLOBAL_CHECKPOINT_S. "
+                "For example, TUNE_GLOBAL_CHECKPOINT_S=60 will perform an "
+                "experiment state snapshot every 60 seconds."
```

Comment:
This recommendation only serves to suppress the warning, by spreading out the experiment state snapshots. It doesn't make experiment snapshotting any faster. 

The scenario where this will be printed is many trials with the default s3 filesystem:
* pyarrow `copy_files(use_threads=True)` hangs for uploading from local to s3. So, we use `use_threads=False` which is really slow. 10+ seconds to upload all directories for ~20-30 trials.
* Even more trials could easily breach the 30s warning threshold.
* The real workaround to decrease syncing time is to switch to s3fs.